controller/konnect/ops/ops.go

```diff
@@ -219,6 +221,19 @@ func Create[
 			time.Since(start),
 		)
 	}
+
+	// For ControlPlane entities, we need to check if the source type is mirror
+	// and set the mirrored condition accordingly.
+	if cp, ok := any(e).(*konnectv1alpha1.KonnectGatewayControlPlane); ok {
+		if isMirrorControlPlane(cp) {
+			if err == nil {
+				SetKonnectEntityMirroredConditionTrue(e)
+			} else {
+				SetKonnectEntityMirroredConditionFalse(e, err)
+			}
+		}
+	}
```

Comment:
Could we put this into an extracted function (like `handleMirrorEntities`)? It will get easier in the future to add handling of new types that support mirror type.

Reply:
I've improved the helpers according to your suggestions. Let me know what you think :)